### PR TITLE
Rename `continue_on` and `start_on` to `continues_on` and `starts_on` with deprecation warnings

### LINKED ISSUE #1410: `continue/start_on` vs `continues/starts_on` naming
P2300 seems to have moved to `continues/starts_on`, while stdexec is using `continue/start_on`. P2300 is correct, no? I'd just like to confirm for compatibility and renamings in pika (which is still calling it `transfer`). If they should be called `continues/starts_on` in stdexec I'd happily do the rename as well here (probably with an alias to `continue/start_on`).

### PR DESCRIPTION
**Pull request by msimberg**
Fix #1410.

Note that I've named the `__starts_on` namespace `__starts_on_ns` because the former would conflict with the `__starts_on` concept (https://github.com/NVIDIA/stdexec/blob/fbd384d1bcbe9dfb9805765f7f40e8dff417a662/include/stdexec/__detail/__transform_sender.hpp#L239). Let me know if you'd prefer it the other way around.

### CODE REVIEW

File: examples/nvexec/maxwell_distributed.cpp [hunk 1]
```diff
@@ -478,13 +478,13 @@ int main(int argc, char *argv[]) {
 
   for (std::size_t compute_step = 0; compute_step < n_iterations; compute_step++) {
     auto compute_h = ex::when_all(
-      ex::just() | exec::on(gpu, ex::bulk(bulk_cells, bulk_h_update)),
-      ex::just() | exec::on(gpu_with_priority, ex::bulk(border_cells, border_h_update))
+      ex::just() | exec::starts_on(gpu, ex::bulk(bulk_cells, bulk_h_update)),
+      ex::just() | exec::starts_on(gpu_with_priority, ex::bulk(border_cells, border_h_update))
```
Comment (ericniebler) [inline examples/nvexec/maxwell_distributed.cpp:482]: code that uses `exec::on` (as opposed to `stdexec::on`) is expecting there-and-back-again behavior, so it should not be changed to `starts_on`. here and elsewhere.

File: README.md [hunk 2]
```diff
@@ -41,9 +41,9 @@ int main()
     // represents the work to later be executed
     auto fun = [](int i) { return i*i; };
     auto work = stdexec::when_all(
-        stdexec::on(sched, stdexec::just(0) | stdexec::then(fun)),
-        stdexec::on(sched, stdexec::just(1) | stdexec::then(fun)),
-        stdexec::on(sched, stdexec::just(2) | stdexec::then(fun))
+        stdexec::starts_on(sched, stdexec::just(0) | stdexec::then(fun)),
+        stdexec::starts_on(sched, stdexec::just(1) | stdexec::then(fun)),
+        stdexec::starts_on(sched, stdexec::just(2) | stdexec::then(fun))
```
Comment (ericniebler) [inline README.md:46]: :+1:

File: include/stdexec/__detail/__execution_fwd.hpp [hunk 3]
```diff
@@ -194,26 +194,36 @@ namespace stdexec {
...
-  extern const continue_on_t continue_on;
+  using transfer_t [[deprecated("transfer_t has been renamed continues_on_t")]] = continues_on_t;
+  [[deprecated("transfer has been renamed continues_on")]]
+  extern const continues_on_t transfer;
 
-  using transfer_t = continue_on_t;
-  extern const transfer_t transfer;
+  using continue_t [[deprecated("continue_on_t has been renamed continues_on_t")]] = continues_on_t;
+  [[deprecated("continue_on has been renamed continues_on")]]
+  extern const continues_on_t continue_on;
```
Comment (ericniebler) [inline include/stdexec/__detail/__execution_fwd.hpp:226]: thanks for the deprecation warnings. :clap:

File: include/stdexec/__detail/__on.hpp [hunk 4]
```diff
@@ -171,9 +171,9 @@ namespace stdexec {
             } else {
               auto&& [__sched, __clsur] = static_cast<_Data&&>(__data);
               return __write_env(                                                       //
-                continue_on(                                                            //
+                continues_on(                                                            //
                   __forward_like<_Data>(__clsur)(                                       //
-                    continue_on(                                                        //
+                    continues_on(                                                        //
```
Comment (ericniebler) [inline include/stdexec/__detail/__on.hpp:176]: pls run clang-format over this pr.

File: test/stdexec/algos/adaptors/test_let_error.cpp [hunk 5]
```diff
@@ -341,7 +341,7 @@ namespace {
       ex::transfer_just(sched3) | ex::let_error([](std::exception_ptr) { return ex::just(); }));
   }
 
-  // Return a different sender when we invoke this custom defined on implementation
+  // Return a different sender when we invoke this custom defined let_error implementation
```
Comment (ericniebler) [inline test/stdexec/algos/adaptors/test_let_error.cpp:344]: nice catch

**Review by ericniebler — COMMENTED** [review]
love this. one comment on `exec::on` but otherwise looks good.

File: examples/nvexec/maxwell_distributed.cpp (hunk 1 above)
Comment (msimberg) [reply examples/nvexec/maxwell_distributed.cpp]: Thanks, good catch. I've removed these (and related) changes in the last batch of changes.

**Comment by msimberg** [comment]
I pushed some more changes that I hope addresses your comments. I missed some renamings in the nvexec directory (I don't have nvhpc set up on my stdexec build at the moment) the first time around. I hope those are all good now as well.

File: include/stdexec/__detail/__on.hpp (hunk 4 above)
Comment (msimberg) [reply include/stdexec/__detail/__on.hpp]: Done. Which version of clang-format do you expect? With 18 I get formatting changes on unrelated parts of the codebase (which I've left out of the changes here), so I was wondering if you expect a specific version. clang-format tends to change its mind about formatting between versions...

Comment (ericniebler) [reply include/stdexec/__detail/__on.hpp]: stdexec has been formatted with clang-format-17. i intend to reformat everything with clang-format-18 as a separate change.

**Pull request merged**